python/smashbox/utilities/__init__.py

```diff
@@ -372,16 +372,28 @@ def run_ocsync(local_folder, remote_folder="", n=None, user_num=None):
         logger.info('sync cmd is: %s',repr(cmd))
 
         process = subprocess.Popen(cmd, shell=False, stdout=logf, stderr=subprocess.STDOUT)
-        process.communicate()
 
-        if process.returncode != 0:
-            msg = "Non-zero exit code %d from command %s" % (process.returncode, repr(cmd))
-            logger.warning(msg)
-            sleep(15)
+        timeout = timeout_min * 60
+
+        while process.poll() is None and timeout > 0:
+            time.sleep(1)
+            timeout -= 1
+
```

Comment:
Instead of the loop, you could actually do:
```
try:
    process.wait(timeout)
except TimeoutExpired:
    pass   # process.poll() will still be None

Reply:
Sadly, [`subprocess.Popen.wait()`](https://docs.python.org/2.7/library/subprocess.html#subprocess.Popen.wait) on 2.7's std lib does not come with support for the `timeout` argument, as opposed to 3.x 😞

Reply:
Oh :( I'm so much in python 3 that I forgot those limitations... Then let's go like this.